Add more metrics in transaction notifier

sadhansood opened the pull request:
As title says, this PR just adds some metrics to track low and high watermarks, and what sequence number batch system is currently at

lxfind commented on crates/sui-core/src/authority_batch.rs:
```diff
@@ -160,6 +160,11 @@ impl crate::authority::AuthorityState {
             .next()
             .unwrap();
 
+        let metrics = self.metrics.clone();
```
Here the code is only called once at node startup, right? Is that intended for this metric?

lxfind reviewed (commented):
A high-level question:
Let's say we redeploy this, and indexer stopped again, there are two cases:
1. low watermark also stopped
2. low watermark kept increasing

Do we have sufficient logs to dig into both cases and find out why?

sadhansood commented:
Take another look @lxfind - I added a couple more things which should be enough imo:
1. batch_svc_is_running - verify if batch service is running
2. debug log to print sequence number of the ticket for the transaction that failed to commit and hence notify
3. We already seem to be logging sequence number in the authority batch loop, so I removed the previous metric batch_service_latest_seq_broadcasted
